### DISCUSSION TITLE
cast SecretStr to str

### DISCUSSION THREAD

**Discussion by sylvainmouquet** [pydantic/pydantic#4217] [Feature Request]
Hello,

To cast a SecretStr we must to use the function get_secret_value() but we cannot use the function str().

```
>>> SecretStr("abc")
SecretStr('**********')

>>> str(SecretStr("abc"))
'**********'

>>> SecretStr("abc").get_secret_value()
'abc'
```

It's for security concerns ?

**Comment by c00kiemon5ter** [comment]
This works as expected.

The `SecretStr` class is used to hide the actual value of the field when printed in logs, dumped on the wire, or serialized for any reason. To get the value you need to invoke the `get_secret_value` method.

**Reply by samuelcolvin** [reply]
This is exactly right, if you don't care about security, just use `str`.

**Comment by sylvainmouquet** [comment]
Yes i will move all my code to `SecretStr`, the implementation of `Secret `provides by Starlette have not the same level of security
